include/networkit/distance/BidirectionalBFS.hpp

```diff
@@ -0,0 +1,45 @@
+/*
+ * BidirectionalBFS.hpp
+ *
+ *  Created on: 14.06.2019
+ *      Author: Eugenio Angriman <angrimae@hu-berlin.de>
+ */
+
+#ifndef BIDIRECTIONAL_BFS_H
+#define BIDIRECTIONAL_BFS_H
+
+#include <networkit/distance/STSP.hpp>
+
+namespace NetworKit {
+
+/*
+ * @ingroup distance
+ * Implements a bidirectional breadth-first search on a graph from two given source and target
+ * nodes. Explores the graph from both the source and target nodes until the two explorations meet.
+ */
+class BidirectionalBFS final : public STSP {
+
+public:
+    using STSP::STSP;
+
+    /*
+     * Perform a bidirectional BFS from the given source and target nodes.
+     */
+    void run() override;
+
+    /*
+     * Returns the distance (i.e., number of hops) from the source to the target node.
+     *
+     * @return count Number of hops from the source to the target node.
+     */
+    count getDistance() {
+        assureFinished();
+        return stDist;
+    }
```

Comment:
What I don't like about this design is that STSP cannot be queried for the distance. I understand that this was done because the BFS distance is an integer. Suggestion: Add a `double getDistance()` method to STSP which works both for Dijkstra and for BFS. Keep this method but rename it to `count getHops()`.